### ISSUE TITLE
关于Loss的疑问

### ISSUE THREAD

**Issue by tulvgengenr** [showlab/Show-o#51]
您好！在`models/modeling_showo.py`文件中，关于Loss的计算代码如下：
```python
loss_t2i = F.cross_entropy(
                logits[:batch_size_t2i, max_seq_length + 1:].contiguous().view(-1, self.output_size),
                labels[:batch_size_t2i, max_seq_length + 1:].contiguous().view(-1), ignore_index=-100,
            )

            # 2. Next token prediction for language modeling
            loss_lm = F.cross_entropy(
                logits[batch_size_t2i:batch_size_t2i + batch_size_lm, :-1].contiguous().view(-1, self.output_size),
                labels[batch_size_t2i:batch_size_t2i + batch_size_lm, 1:].contiguous().view(-1), ignore_index=-100,
            )

            # 3. Next token prediction for captioning/multimodal understanding
            loss_mmu = F.cross_entropy(
                logits[-batch_size_mmu:, :-1].contiguous().view(-1, self.output_size),
                labels[-batch_size_mmu:, 1:].contiguous().view(-1), ignore_index=-100,
```
我有一个疑惑，在t2i任务中，使用的是logits[:batch_size_t2i, max_seq_length + 1:]和labels[:batch_size_t2i, max_seq_length + 1:]计算交叉熵loss，这貌似表示在预测image的时候，logits不再表示next token的概率，而就是当前token
的概率。并没有像lm和mmu任务中，logits和labels错位1。

**Comment by Sierkinhane** [comment]
image generation我们是采用了discrete diffusion（or mask token prediction），具体细节可以看文章的preliminary和方法部分哈
